build-tools/manifest-attribute-codegen/Models/ManifestDefinition.cs

```diff
@@ -0,0 +1,87 @@
+using System.Xml.Linq;
+
+namespace Xamarin.Android.Tools.ManifestAttributeCodeGenerator;
+
+class ManifestDefinition
+{
+	public string ApiLevel { get; set; } = "0";
+	public List<ElementDefinition> Elements { get; } = new List<ElementDefinition> ();
+
+	// Creates a new ManifestDefinition for a single Android API from the given file path
+	public static ManifestDefinition FromFile (string filePath)
```

Comment:
May want to mention that `filePath` is expected to be similar to `$HOME/android-toolchain/sdk/platforms/android-34/data/res/values/attrs_manifest.xml`, and thus `dir_name` would be `android-34`, and `elements` + `//declare-styleable` is coming from `attrs_manifest.xml`.